src/seedu/addressbook/parser/Parser.java

```diff
@@ -228,6 +259,25 @@ private int parseArgsAsDisplayedIndex(String args) throws ParseException, Number
         return Integer.parseInt(matcher.group("targetIndex"));
     }
 
+    /**
+     * Parses the given arguments string as a list of index numbers.
+     *
+     * @param args arguments string to parse as a list of index numbers
+     * @return the parsed list of index numbers
+     * @throws ParseException if no region of the args string could be found for the index
+     * @throws NumberFormatException the args string region is not a valid number
+     */
+    private String[] parseArgsAsDisplayedIndices(String args) throws ParseException, NumberFormatException {
+        final String[] extractedIndices = args.trim().split(" ");
+        for (String index: extractedIndices) {
+            if (!index.matches("\\d+")){
```

Comment:
Regular expression is not cheap to compute, one way to check whether it is a number is to cast a string to number and if it raises an exception, it is not a number.